modular_skyrat/modules/borg_buffs/code/snack_dispensor.dm

```diff
@@ -0,0 +1,142 @@
+	. = ..()
+	if(.)
+		var/obj/item/borg_snack_dispensor/snack_dispensor = new /obj/item/borg_snack_dispensor(R.model)
+		R.model.basic_modules += snack_dispensor
+		R.model.add_module(snack_dispensor, FALSE, TRUE)
+		for(var/obj/item/rsf/cookiesynth/cookiesynth in R.model)
+			removed_modules += cookiesynth
+			R.model.remove_module(cookiesynth)
+		for(var/obj/item/borg/lollipop/lollipop in R.model)
+			removed_modules += lollipop
+			R.model.remove_module(lollipop)
+
+/obj/item/borg/upgrade/snack_dispensor/deactivate(mob/living/silicon/robot/R, user)
+	. = ..()
+	if(!.)
+		return
+	for(var/obj/item/borg_snack_dispensor/dispensor in R.model)
+		R.model.remove_module(dispensor, TRUE)
+	for(var/obj/item as anything in removed_modules)
+		R.model.basic_modules += item
+		R.model.add_module(item, FALSE, TRUE)
+
+/obj/item/borg_snack_dispensor
+	name = "\improper Automated Borg Snack Dispensor"
+	desc = "Has the ability to automatically print many differnt forms of snacks. Now Lizard approved!"
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rsf"
+	/// Contains the PATH of the selected snack
+	var/atom/selected_snack
+	/// Whether snacks are launched when targeted at a distance
+	var/launch_mode = FALSE
+	/// A list of all valid snacks
+	var/list/valid_snacks = list(
+		/obj/item/food/cookie/bacon,
+		/obj/item/food/cookie/cloth,
+		/obj/item/food/cookie/sugar,
+		/obj/item/food/lollipop/cyborg
+	)
+	/// Minimum amount of charge a borg can have before snack printing is disallowed
+	var/borg_charge_cutoff = 200
+	/// The amount of charge used per print of a snack
+	var/borg_charge_usage = 50
+
+/obj/item/borg_snack_dispensor/Initialize(mapload)
+	. = ..()
+	selected_snack = selected_snack ||  LAZYACCESS(valid_snacks, 1)
+
+/obj/item/borg_snack_dispensor/examine(mob/user)
+	. = ..()
+	var/snack_name = initial(selected_snack.name)
+	. += "It is currently set to dispense [snack_name]."
+	. += "You can AltClick it to [(launch_mode ? "disable" : "enable")] launch mode."
+
+/obj/item/borg_snack_dispensor/attack_self(mob/user, modifiers)
+	var/list/choices = list()
+	for(var/atom/snack as anything in valid_snacks)
+		choices[initial(snack.name)] = snack
+	if(!length(choices))
+		to_chat(user, span_warning("No valid snacks in database."))
+	if(length(choices) == 1)
+		selected_snack = choices[1]
+	else
+		var/selected = tgui_input_list(user, "Select Snack", "Snack Selection", choices)
+		if(!selected)
+			return
+		selected_snack = choices[selected]
+	var/snack_name = initial(selected_snack.name)
+	to_chat(user, span_notice("[src] is now dispensing [snack_name]"))
+
+/obj/item/borg_snack_dispensor/attack(mob/living/patron, mob/living/user, params)
+	var/empty_hand = LAZYACCESS(patron.get_empty_held_indexes(), 1)
+	if(!empty_hand)
+		to_chat(user, span_warning("[patron] has no free hands!"))
+		return
+	if(!selected_snack)
+		to_chat(user, span_warning("No snack selected."))
+		return
+	var/mob/living/silicon/robot/borg = user
```

Comment:
You could change the typecasting of the user argument instead of making a new var